[Teo Kai Sheng] iP

teoks0199 opened the pull request:
### Duke Task Manager :smiling_imp:
- text-based task manager
- easy to learn

All you need to do is 
1. download it [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :kissing_closed_eyes:

Features
- [x] Adding and removing tasks
- [x] Marking tasks
- [ ] Reminders *(coming soon)*

Some commands you can use:

> todo get milk
> deadline return book /by 2023-09-09
> event exams /from 2023-10-10 /to 2023-10-21
> mark 1
> list
> delete 1

Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke(Paths.get(".", "data", "duke.txt")).run(); // ./data/duke.txt
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

teoks0199 committed:
Rename, Greet, Exit

teoks0199 committed:
Echo

teoks0199 committed:
Add, List

teoks0199 committed:
Mark as Done

teoks0199 committed:
ToDos, Events, Deadlines

teoks0199 committed:
Automated Text UI Testing

teoks0199 committed:
Handle Errors

teoks0199 committed:
Delete

teoks0199 committed:
add javadocs

teoks0199 committed:
Save

teoks0199 committed:
test

teoks0199 committed:
Merge branch 'branch-Level-7'

teoks0199 committed:
Dates and Times

teoks0199 committed:
Merge branch 'branch-Level-8'

teoks0199 committed:
More OOP

teoks0199 committed:
Merge branch 'branch-A-MoreOOP'

teoks0199 committed:
Add packages

teoks0199 committed:
Fix bugs

teoks0199 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

teoks0199 committed:
Add gradle

teoks0199 committed:
Fix gradlew run application path

teoks0199 committed:
Merge branch 'branch-A-Gradle'

teoks0199 committed:
Add junit testing

teoks0199 committed:
Add saved test list for testing

teoks0199 committed:
Add Jar

teoks0199 committed:
Merge branch 'branch-A-jar'

teoks0199 committed:
Add javadocs

teoks0199 committed:
Fix style errors

teoks0199 committed:
Add find functionality

teoks0199 committed:
Merge branch 'branch-A-JavaDoc'

teoks0199 committed:
Merge branch 'branch-A-CodingStandard'

teoks0199 committed:
Merge branch 'branch-Level-9'

teoks0199 committed:
Fix style

lululwtv commented on text-ui-test/runtest.bat:
```diff
@@ -7,7 +7,7 @@ REM delete output from previous run
 
 REM compile the code into the bin folder
-javac  -cp ..\src\main\java -Xlint:none -d ..\bin ..\src\main\java\*.java
+javac  -cp ..\src\main\java -Xlint:none -d ..\bin ..\src\main\java\duke\*.java
```
I like how you are still using automated testing!

lululwtv commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,19 @@
+todo borrow book
+list
+deadline return book /by Sunday
+deadline return book /by 2023-08-11
+event project meeting /from 2023-08-11 /to 2023-08-19
+unmark 2
+mark 2
+list
+mark 1
+unmark 2
+todo eat
+delete 1
+list
+delete 1
+delete 1
+delete 1
+mark 1
+delete 1
+bye
```
I like the thorough testing done

lululwtv commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,67 @@
+    private TaskList tasks;
+    private boolean finish;
+
+    /**
+     * Constructor to create a Parser.
+     *
+     * @param ui Instance to handle the printing of user interface.
+     * @param tasks Instance of TaskList to handle commands.
+     */
```
I like your documentation! Very clear!

teoks0199 committed:
Change names

teoks0199 committed:
Add checkstyle

teoks0199 committed:
Merge branch 'branch-A-Checkstyle'

jedkohjk commented on src/main/java/duke/Duke.java:
LGTM

jedkohjk commented on src/main/java/duke/DukeException.java:
LGTM

jedkohjk commented on src/main/java/duke/Parser.java:
LGTM

jedkohjk commented on src/main/java/duke/Storage.java:
LGTM

jedkohjk commented on src/main/java/duke/TaskList.java:
LGTM

jedkohjk commented on src/main/java/duke/Ui.java:
LGTM

jedkohjk commented on src/main/java/duke/task/Deadline.java:
LGTM

jedkohjk commented on src/main/java/duke/task/Event.java:
LGTM

jedkohjk commented on src/main/java/duke/task/Task.java:
LGTM

jedkohjk commented on src/main/java/duke/task/ToDo.java:
LGTM

jedkohjk commented on src/test/java/duke/StorageTest.java:
LGTM

jedkohjk commented on src/test/java/duke/TaskListTest.java:
LGTM

jedkohjk commented on src/test/testlist.txt:
LGTM

jedkohjk reviewed (commented):
Overall, I found your code easy to read. LGTM.

teoks0199 committed:
Add fxml

teoks0199 committed:
Fix linkage error

teoks0199 committed:
Add UI

teoks0199 committed:
Merge branch 'branch-Level-10'

teoks0199 committed:
JAR file: for command line interface version

Program has since adopted a GUI version.

Let's make a new updated build for the GUI version.

The older JAR file is already outdated.

teoks0199 committed:
Update javadocs

teoks0199 committed:
Update javadocs

teoks0199 committed:
Fix list not saving

teoks0199 committed:
Add assertions

Duke class has no assertions.

Some errors in the program may not be caught.

Let's implement assertions at various points in the code.

This will check if important assumptions at those points hold true.

teoks0199 committed:
Add named constants

Initial welcome message is a magic string.

It is confusing what it really means.

Let's assign it to a named constant.

The name will tell us the meaning of the string.

teoks0199 committed:
Merge pull request #1 from teoks0199/branch-A-CodeQuality

Add named constants

teoks0199 committed:
Merge branch 'master' of https://github.com/teoks0199/ip

teoks0199 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Duke.java

teoks0199 committed:
Merge pull request #3 from teoks0199/branch-A-Assertions

Branch a assertions

teoks0199 committed:
Merge branch 'master' of https://github.com/teoks0199/ip

teoks0199 committed:
Fix code quality further.

TaskList class: Handle the printing of error messages.

Many magic strings are being used for these error messages.

Let's move the printing of error messages to the Ui class.

TaskList will no longer have to use magic literals to print out
error messages, making the code clearer.

teoks0199 committed:
Setup continuous integration

teoks0199 committed:
Add update functionality

teoks0199 committed:
Merge branch 'branch-C-Update'

teoks0199 committed:
Add background color

teoks0199 committed:
Merge branch 'branch-A-BetterGui'

teoks0199 committed:
Add javadocs

teoks0199 committed:
Add user guide

teoks0199 committed:
Fix fxml version

teoks0199 committed:
Remove unused implementations from build.gradle

teoks0199 committed:
Add acceptable values to user guide

teoks0199 committed:
Add more GFMD in user guide
